paperqa/agents/task.py

```diff
@@ -487,3 +493,251 @@ def __len__(self) -> int:
     LitQAv2TaskDataset.__module__,
     LitQAv2TaskDataset.__name__,
 )
+
+
+class LFRQAPairwiseEvalEnv(GradablePaperQAEnvironment):
+    def __init__(
+        self,
+        *args,
+        qid: str | UUID,
+        question: str,
+        human_answer: str,
+        gt_doc_ids: list[str],
+        pairwise_eval_llm: LLMModel | str = CommonLLMNames.GPT_4O.value,
+        **kwargs,
+    ):
+        # NOTE I'm using qid and question and
+        # maybe we could use session_id and query instead?
+        kwargs["query"] = question
+
+        # NOTE: I'm also passing this docs because they are required
+        # but don't know if/how they are used
+        kwargs["docs"] = Docs()
+        super().__init__(*args, **kwargs)
+
+        self.qid = qid
+        self.question = question
+        self.human_answer = human_answer
+        self.gt_doc_ids = gt_doc_ids
+        self.pairwise_eval_llm = pairwise_eval_llm
+
+    def extract_best_answer_index(self, text: str) -> int:
+        match = re.search(r"<rating>(\d+)</rating>", text)
```

Comment:
This isn't quite the most general XML extraction logic, here's a more general way: https://stackoverflow.com/a/13241658

Also it might not matter, just sharing

Reply:
I think the difference is that `<rating>(\d+)</rating>` only searches for numbers, and that's ok in this case.